local/rebac-admin/qa.sh

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env bash
+# Copyright (C) 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+set -e
+
+_usage='Exercise running server
+
+Options:
+  --help           displays current doc.
+  --reset          resets the server state at the beginning; default is false.
+  --cleanup        cleans up created entities/relationships at the end; default is false.
+  --bail-on-error  exits when a request fails (status code >= 400); default is false.
+'
+
+_host="localhost:17070"
+_base="$_host/rebac/v1"
+
+generate_random_string() {
+    local length=$1
+    tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w "$length" | head -n 1
+}
+num_strings=5
+
+# Length of each random string
+string_length=8
+
+# Generate array with random unique strings
+_names=()
+while [[ ${#_names[@]} -lt $num_strings ]]; do
+    new_string=$(generate_random_string "$string_length")
+    if [[ ! " ${_names[*]} " =~ " ${new_string} " ]]; then
+        _names+=("$new_string")
+    fi
+done
+
+# _names=(alpha bravo charlie delta echo foxtrot golf hotel india juliet kilo lima mike november oscar papa quebec romeo sierra tango uniform victor whiskey x yankee zulu)
+
+_reset_at_start=""
+_cleanup_at_end=""
+_bail_on_error=""
+_help=""
+for option in "$@" ; do
+    if [[ "$option" == "--reset" ]] ; then
+        _reset_at_start="true"
+    fi
+    if [[ "$option" == "--cleanup" ]] ; then
+        _cleanup_at_end="true"
+    fi
+    if [[ "$option" == "--bail-on-error" ]] ; then
+        _bail_on_error="true"
+    fi
+    if [[ "$option" == "--help" || "$option" == "-h" ]] ; then
+        _help="true"
+        break
+    fi
+done
+
+if [ "$_help" == "true" ]; then
+    echo "$_usage"
+    exit 0
+fi
+
+## Check if the server is running
+if ! curl -s "$_host/health"; then
+    function onexit {
+        curl "$_host/shutdown"
+        wait $_PID1
+        echo "Server shut down"
+    }
+    trap onexit EXIT
+
+    ## Run server in background
+    go build -o server ./cmd
+    bash -c 'sleep 1 && ./server' &
```

Comment:
Is this used? I don't see how it would work without setting a bunch of env vars for Postgres, Vault, etc.

Reply:
this is to be removed